chore: update label to use tailwind css

raastinn opened the pull request:
## Changes
<!--- Provide the broad strokes of the changes made in a bulleted list --->
- updated `Label` component to use Tailwind
- Removed `theme` prop from `getBadgeOrLabelColor` since it wasn't necessary
- Fixed tailwind styling not being applied
- Removed spacing prop and did the styling in parent component

## Compatiblity

Any changes that require a newer version of the API are considered breaking
changes.

Common breaking changes:
 - Making a request to a new API endpoint that will return `403` or `404` in older versions of the API.
 - No longer sending request fields required by previous versions of the API 
 - Requiring new fields from an API endpoint that will not be present in
   responses from older versions of the API
 - Breaking backwards compatibility with old response data shapes


Choose one:
 - [x] The changes do not break compatibility
 - [ ] The changes potentially break compatibility


## Pre-Review Checklist
 - [x] All changes are tested
 - [x] All touched code documentation is updated
 - [x] All tests pass in your local environment
 - [x] Deepsource issues have been reviewed and addressed
 - [x] Debugging logging and commented out code has been removed

 
## Screenshots:
_Only required for visual changes_.

raastinn committed:
chore: update label to use tailwind css

igboyes commented on src/index.html:
```diff
@@ -30,6 +30,7 @@
 
         <link rel="shortcut icon" href="/images/favicon.ico" type="image/x-icon" />
         <script type="module" src="./js/index.tsx"></script>
+        <script src="https://cdn.tailwindcss.com"></script>
```
This shouldn't be necessary.

The styles and logic should be bundled in the app based off your Tailwind installation and config.

Did you follow the Vite guide? https://tailwindcss.com/docs/guides/vite

raastinn replied:
Yeah, I followed the guide but the styling wouldn't apply. It turns out the create sample file selector styling was because of the styled version of `CompactScrollList`. I tried some other methods too but this one was the only one that seemed to apply the styling changes

igboyes commented on src/js/account/components/AccountProfile.tsx:
```diff
@@ -56,9 +48,13 @@ export default function AccountProfile() {
                         {administrator_role && (
-                            <AdministratorTag key="administrator" color="purple">
+                            <Label
+                                className={`capitalize text-[${theme.fontSize.md}] ml-auto`}
```
Tailwind is the theme.
```suggestion
                                className="capitalize text-md ml-auto"
```

igboyes commented on src/js/base/Label.tsx:
```diff
@@ -0,0 +1,26 @@
+            }] ${className} align-baseline last-of-type:m-0 `}
+        >
+            {children}
+        </span>
```
* Just use the Tailwind classes like `text-white`, `rounded-sm` directly.
* Use something like `cslx` or `classnames` packages for conditional styles. Try `cslx` first if one isn't already installed.

igboyes commented on src/js/base/NotFound.tsx:
```diff
@@ -27,7 +24,9 @@ type NotFoundProps = {
     return (
         <StyledNotFound>
-            <Label color="red">{status}</Label>
+            <Label className={`text-[${theme.fontSize.xxl}]`} color="red">
```
Tailwind is (will be) our theme.

Make sure the `text-` suffix here is similar to what it was before in appearance. Like, you might have to use `xl` instead.
```suggestion
            <Label className="text-xxl" color="red">
```

raastinn committed:
fix: tailwind config & label styling, install clsx

raastinn committed:
chore: update label

raastinn committed:
refactor: rework components using Label to use class names

raastinn committed:
refactor: add cn function to merge tailwind styling

raastinn commented:
- I moved our colours to the tailwind theme configs since its much more simplified than the theme used by tailwind (they use bg-red-50, bg-red-100, bg-red 200, etc) whereas now its just simplified to `bg-red`, `bg-red-light`.

- Also removed the `color` prop from the `Label` component if we're going full tailwind.

Let me know your thoughts

igboyes commented:
@raastinn

* Restore `color` prop for `Label`. It will set `className` based on the color variant (eg. "red") provided.
* Reset Tailwind theming to default.
* You can use my changes or drop them and work off yours. I started doing this because I needed it done for something else I was doing. Then, I got pulled into something else and this is stagnant.

Thanks!

raastinn committed:
refactor: use default tailwind theme

igboyes commented on src/js/analyses/components/Iimi/IimiDetectionTag.tsx:
```diff
@@ -1,6 +1,6 @@
 
 export function IimiDetectionTag({ result }) {
-    return result ? <Label color="red">Detected</Label> : <Label color="grey">Undetected</Label>;
+    return result ? <Label color="red">Detected</Label> : <Label className="bg-grey">Undetected</Label>;
```
Can't this just be like below since grey is the default?:
```suggestion
    return result ? <Label color="red">Detected</Label> : <Label>Undetected</Label>;
```

igboyes commented on src/js/hmm/components/HMMDetail.tsx:
```diff
@@ -120,7 +116,9 @@ export default function HMMDetail({ match }: HMMDetailProps) {
-                            <td>{names}</td>
+                            <td>
+                                <div className="flex gap-1 leading-none">{names}</div>
+                            </td>
```
Does this not work?
```suggestion
                            <td className="flex gap-1 leading-none">
                                {names}
                            </td>
```

raastinn replied:
That doesn't work but I just applied the styling to the label instead

igboyes reviewed (changes requested):
Very close.

raastinn committed:
Merge branch 'main' into rastinrashidi/ui-158-update-label-base-component

raastinn committed:
chore: minor tweaks

ghost commented:
:tada: This issue has been resolved in version 6.3.0 :tada:

The release is available on [GitHub release](https://github.com/virtool/virtool-ui/releases/tag/6.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
